alf/environments/suite_babyai.py

```diff
@@ -186,15 +187,19 @@ def __init__(self,
 
     def _tokenize(self, instruction):
         """Convert instruction string to a numpy array."""
+        if self._mode == 'char':
+            tokens = np.array([ord(c) for c in instruction])
+            return tokens
+
         tokens = self._word_pattern.findall(instruction.lower())
-        instr = np.array([self._vocab.get(token, 0) for token in tokens])
-        if np.amin(instr) == 0:
+        tokens = np.array([self._vocab.get(token, 0) for token in tokens])
+        if np.amin(tokens) == 0:
```

Comment:
This is an unsafe rename -- collides with the line below which also references "tokens" as the words before being translated into word_ids.
Let's name it token_ids or something like that?